x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/action_connector_form/test_connector_form.tsx

```diff
@@ -181,25 +183,56 @@ const AwaitingExecution = () => (
   </EuiCallOut>
 );
 
-const SuccessfulExecution = () => (
-  <EuiCallOut
-    title={i18n.translate(
-      'xpack.triggersActionsUI.sections.testConnectorForm.executionSuccessfulTitle',
-      {
-        defaultMessage: 'Test was successful',
-      }
+const SuccessfulExecution = ({
+  executionResult,
+}: {
+  executionResult: ActionTypeExecutorResult<unknown> | undefined;
+}) => (
+  <>
+    <EuiCallOut
+      title={i18n.translate(
+        'xpack.triggersActionsUI.sections.testConnectorForm.executionSuccessfulTitle',
+        {
+          defaultMessage: 'Test was successful',
+        }
+      )}
+      color="success"
+      data-test-subj="executionSuccessfulResult"
+      iconType="check"
+    >
+      <p>
+        <FormattedMessage
+          defaultMessage="Ensure the results are what you expect."
+          id="xpack.triggersActionsUI.sections.testConnectorForm.executionSuccessfulDescription"
+        />
+      </p>
+    </EuiCallOut>
+    {executionResult && (
+      <>
+        <EuiSpacer size="s" />
+        <EuiTitle size="xs">
+          <h4>
+            {i18n.translate(
+              'xpack.triggersActionsUI.sections.testConnectorForm.executionResultDetails',
+              {
+                defaultMessage: 'Response',
+              }
+            )}
+          </h4>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        <EuiCodeBlock
+          language="json"
+          paddingSize="m"
+          overflowHeight={300}
+          isCopyable
+          data-test-subj="executionResultCodeBlock"
+        >
+          {JSON.stringify(executionResult, null, 2)}
```

Comment:
Let's try/catch and silently fail this, just in case the response is malformed.

```
const getExecutionResultResponse = (executionResult) => {
  try {
    return JSON.stringify(executionResult, null, 2)
  } catch {
    return null
  }
}

<EuiCodeBlock
          language="json"
          paddingSize="m"
          overflowHeight={300}
          isCopyable
          data-test-subj="executionResultCodeBlock"
        >
          {getExecutionResultResponse(executionResult)}
</EuiCodeBlock>
```